src/basecases.py

```diff
@@ -60,6 +61,54 @@ def fiveDummyItemNodes():
     return itemnodes
 
 
+def bootstrap_dependencies_three_nodes():
+    """
+    initialize demo_node with the following dependencies
+    0 -> 1 -> 2
+    """
+    demo_nodes = [
+        SingleItemNode(node_id=i, dependency=items.ItemDependency([], "")) for i in range(3)
+    ]
+
+    wood = items.ItemReq(items.Item('wood', 0), 1)
+    door = items.ItemReq(items.Item('door', 1), 1)
+    house = items.ItemReq(items.Item('house', 2), 1)
+
+    demo_nodes[0].dependency = items.ItemDependency([], wood)
+    demo_nodes[1].dependency = items.ItemDependency([wood], door)
+    demo_nodes[2].dependency = items.ItemDependency([door], house)
+
+    return demo_nodes
+
+
+def bootstrap_dependencies_five_nodes():
+    """
+    initialize demo_node with the following dependencies
+    0 -> | ->  1 -----> | -> 3 -> | -> 4 -> | -> 5
+           ->  2 -> |-> |         |
+                 -> |-----------> |
+    """
+    demo_nodes = [
+        SingleItemNode(node_id=i, dependency=items.ItemDependency([], "")) for i in range(6)
+    ]
+
+    start = items.ItemReq(items.Item('start', 0), 1)
+    wood = items.ItemReq(items.Item('wood', 1), 1)  # There is always only one node starting the whole graph!
+    timber = items.ItemReq(items.Item('timber', 2), 1)
+    premium_timber = items.ItemReq(items.Item('premium_timber', 3), 1)
+    door = items.ItemReq(items.Item('door', 4), 1)
+    house = items.ItemReq(items.Item('house', 5), 1)
```

Comment:
we might want better names and IDs to look "pretty"/realistic. e.g. Item(door-type-x-blue, XYT65)